portal/src/main/webapp/js/src/oncoprint/webgl/src/js/oncoprintruleset.js

```diff
@@ -472,11 +472,17 @@ var LinearInterpRuleSet = (function () {
 	this.value_key = params.value_key;
 	this.value_range = params.value_range;
 	this.log_scale = params.log_scale; // boolean
-	this.inferred_value_range;
+
+	this.rangeTypes = {
+		'SIGNED': 'SIGNED',
+		'UNSIGNED': 'UNSIGNED',
+		'NON_POSITIVE': 'NON_POSITIVE'
```

Comment:
@pieterlukasse exactly what @adamabeshouse said. I used the term non-positive to implies that it includes 0.

I would also go with Adam's suggestion but I'm afraid it will mislead the inclusion of 0's.

What about:
- NON_POSITIVE = negative values inclusive 0
- NON_NEGATIVE = positive values inclusive 0
- ALL